app/src/main/java/com/github/sdp_begreen/begreen/utils/TinyDB.kt

```diff
@@ -0,0 +1,102 @@
+package com.github.sdp_begreen.begreen.utils
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.text.TextUtils
+import androidx.preference.PreferenceManager
+import com.google.gson.Gson
+import com.google.gson.JsonParser
+import java.util.*
+
+// This class is a utility to simplify the use of SharedPreferences for Android.
+class TinyDB(appContext: Context) {
```

Comment:
I think you should cite your source when you copy past a file...

https://github.com/kcochibili/TinyDB--Android-Shared-Preferences-Turbo/blob/master/TinyDB.java#L287

Reply:
Indeed, you are right! I should cite the most possible when I take existing code even if I made changes on it. However, as the code is not functioning and not yet well written in terms of methods declaration and comments, I did not consider it as urgent as of now. But I agree I should take the habit to do it at the very beginning of coding to follow good standard practice and save time later on. Thanks for pointing that out!